From cc2b7884986c706d622d5fef4a5f47fe98ca7b02 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:38:43 -0400
Subject: [ethernet] i40e: don't complain about removing non-existent addresses

Message-id: <1409658042-22689-163-git-send-email-sassmann@redhat.com>
Patchwork-id: 90357
O-Subject: [RHEL7.1 PATCH 162/281] i40e: don't complain about removing non-existent addresses
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shannon Nelson <shannon.nelson@intel.com>

We don't need to complain in the log about mac addresses that
can't be deleted because they don't exist.

Change-ID: I4e6370df175bf72726f06d2206c03bcbfded8387
Signed-off-by: Shannon Nelson <shannon.nelson@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit fdfe9cbe0f002af8fa94c04639d2f224be9847d5)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 35e79e4..60022c7 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -1595,7 +1595,9 @@ int i40e_sync_vsi_filters(struct i40e_vsi *vsi)
     num_del = 0;
     memset(del_list, 0, sizeof(*del_list));
 
-    if (aq_ret)
+    if (aq_ret &&
+        pf->hw.aq.asq_last_status !=
+             I40E_AQ_RC_ENOENT)
      dev_info(&pf->pdev->dev,
        "ignoring delete macvlan error, err %d, aq_err %d while flushing a full buffer\n",
        aq_ret,
@@ -1607,7 +1609,8 @@ int i40e_sync_vsi_filters(struct i40e_vsi *vsi)
            del_list, num_del, NULL);
    num_del = 0;
 
-   if (aq_ret)
+   if (aq_ret &&
+       pf->hw.aq.asq_last_status != I40E_AQ_RC_ENOENT)
     dev_info(&pf->pdev->dev,
       "ignoring delete macvlan error, err %d, aq_err %d\n",
       aq_ret, pf->hw.aq.asq_last_status);
-- 
1.7.1